neqo-common/src/datagram.rs

```diff
@@ -9,23 +9,20 @@ use std::{net::SocketAddr, ops::Deref};
 use crate::{hex_with_len, IpTos};
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct Datagram {
+pub struct Datagram<D = Vec<u8>> {
     src: SocketAddr,
     dst: SocketAddr,
+    /// The size of each segment within the [`Datagram`]. All segments, but the
+    /// last, have the same size. The last segment can be shorter than
+    /// [`Datagram::segment_size`].
+    segment_size: usize,
     tos: IpTos,
-    d: Vec<u8>,
+    d: D,
 }
 
-impl Datagram {
-    pub fn new<V: Into<Vec<u8>>>(src: SocketAddr, dst: SocketAddr, tos: IpTos, d: V) -> Self {
-        Self {
-            src,
-            dst,
-            tos,
-            d: d.into(),
-        }
-    }
+impl Copy for Datagram<&[u8]> {}
```

Comment:
Really?  That's a pretty large thing to throw around like that.  I might prefer not to have it implement Copy.  I understand why this might be convenient, but that convenience hides actual work.  A Datagram will be at least 59 bytes in size, plus adjustments for alignment.